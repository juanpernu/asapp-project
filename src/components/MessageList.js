/** React dependencies */
import React from "react";

/** Component dependencies */
import Message from './Message';

/**
 * Stateless Component that draw the box where
 * the messages are listed.
 * @param {String} messages
 * @param {Object} user
 */
const MessageList = ({ messages, user }) => {
  return (
    <div className="message-list">
      {messages.map((message, index) => (
        <Message
          key={index}
          message={message.text}
          from={message.from}
          user_id={user.id}
          timestamp={message.timestamp}
          saw_message={message.saw_message}/>)
      )}
    </div>
  )
}

export default MessageList;
