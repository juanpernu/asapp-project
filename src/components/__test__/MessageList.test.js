import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import MessageList from '../MessageList';

const mockedData = {
  messages: [{
    text: "Text message",
    from: {
      name: "Rob",
      id: 0,
      image_profile: "https://images.unsplash.com/photo-1487529236533-f2b78c0c9344?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&rect=1664%2C1838%2C1033%2C1033&fit=crop&s=9a5a069600d20a4718f41bce3aec4225&h=128&w=128"
    },
    timestamp: "12:04",
    user_id: 0,
    saw_message: false
  }],
  user: {
    name: "Rob",
    id: 0,
    image_profile: "https://images.unsplash.com/photo-1487529236533-f2b78c0c9344?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&rect=1664%2C1838%2C1033%2C1033&fit=crop&s=9a5a069600d20a4718f41bce3aec4225&h=128&w=128",
  }
}

describe('Componentes - MessageList', () => {
  it('Should render correctly', () => {
    window.HTMLElement.prototype.scrollIntoView = function () { };
    render( <MessageList messages={mockedData.messages} user={mockedData.user}/>);
    expect(MessageList).toMatchSnapshot();
  });
});
