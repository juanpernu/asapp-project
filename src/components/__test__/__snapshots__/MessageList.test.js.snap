// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componentes - MessageList Should render correctly 1`] = `[Function]`;
